src/main/scala/com/mongodb/spark/rdd/partitioner/MongoSplitVectorPartitioner.scala

```diff
@@ -64,15 +64,19 @@ class MongoSplitVectorPartitioner extends MongoPartitioner {
     val partitionSize = partitionerOptions.getOrElse(partitionSizeMBProperty, DefaultPartitionSizeMB).toInt
 
     val keyPattern: BsonDocument = new BsonDocument(partitionKey, new BsonInt32(1))
+    val minKeyMaxKey = PartitionerHelper.getRangeFilterFromPipeline(partitionKey, pipeline)
```

Comment:
The split vector partitioner uses `PartitionerHelper.getRangeFilterFromPipeline` to calculate the min and max values for the command.

It also ensures the correct boundaries are applied when creating the partitioners - this stops the partitions expanding to include a `minKey` and `maxKey` boundary at the start and end.